hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializerTest.java

```diff
@@ -529,50 +613,98 @@ public void testSchemaEvolution_GenericRecord() {
     }
 
     @Test
-    public void testSchemaEvolution_fieldAdded() {
-        SerializationConfig serializationConfig = new SerializationConfig();
-        //Using this registration to mimic schema evolution. This is usage is not advised.
-        serializationConfig.getCompactSerializationConfig()
-                .addSerializer(new CompactSerializer<EmployeeDTO>() {
-                    @Nonnull
-                    @Override
-                    public EmployeeDTO read(@Nonnull CompactReader reader) {
-                        throw new UnsupportedOperationException("We will not read from here on this test");
-                    }
+    public void testSchemaEvolution_variableSizeFieldAdded() {
+        //Using this type of serializer to mimic schema evolution. This is usage is not advised.
+        CompactSerializer<EmployeeDTO> serializer = new CompactSerializer<EmployeeDTO>() {
+            @Nonnull
+            @Override
+            public EmployeeDTO read(@Nonnull CompactReader reader) {
+
+                int age = reader.readInt32("age");
+                long id = reader.readInt64("id");
+                String surname = "N/A";
+                if (reader.getFieldKind("surname") == FieldKind.STRING) {
+                    surname = reader.readString("surname");
+                }
+                return new EmployeeDTO(age, id);
+            }
+
+            @Override
+            public void write(@Nonnull CompactWriter writer, @Nonnull EmployeeDTO object) {
+                writer.writeInt32("age", object.getAge());
+                writer.writeInt64("id", object.getId());
+                writer.writeString("surname", "sir");
+            }
+
+            @Nonnull
+            @Override
+            public String getTypeName() {
+                return EmployeeDTO.class.getName();
+            }
+
+            @Nonnull
+            @Override
+            public Class<EmployeeDTO> getCompactClass() {
+                return EmployeeDTO.class;
+            }
+        };
+
+        SerializationService serializationService = createSerializationService(() -> serializer);
 
-                    @Override
-                    public void write(@Nonnull CompactWriter writer, @Nonnull EmployeeDTO object) {
-                        writer.writeInt32("age", object.getAge());
-                        writer.writeInt64("id", object.getId());
-                        writer.writeString("surname", "sir");
-                    }
+        EmployeeDTO expected = new EmployeeDTO(20, 102310312);
+        Data data = serializationService.toData(expected);
 
-                    @Nonnull
-                    @Override
-                    public String getTypeName() {
-                        return EmployeeDTO.class.getName();
-                    }
+        // Assert that older client can read newer data
+        SerializationService serializationService2 = createSerializationService();
```

Comment:
The changes look good to me. Maybe we could use `newSerializationService`, `oldSerializationService` etc. to distinguish serialization services in these kinds of tests to make them more clear